data-prepper-plugins/s3-source/src/main/java/org/opensearch/dataprepper/plugins/source/s3/ScanObjectWorker.java

```diff
@@ -209,6 +215,17 @@ private void startProcessingObject(final long waitTimeMillis) {
                     }
                     partitionKeys.remove(objectToProcess.get().getPartitionKey());
                 }, ACKNOWLEDGEMENT_SET_TIMEOUT);
+
+                acknowledgementSet.addProgressCheck(
```

Comment:
This approach only works for end-to-end acknowledgements enabled. The root cause is an issue writing to the buffer, not a failure to write to the sink. So this change may actually hide the bug some.

Reply:
Yes it does only resolve the issue for acknowledgments being enabled. But this aligns with how we handle visibility timeout increase on the S3-SQS source.

Reply:
We should correct both of these really. But, this is fine for now.

Reply:
@graytaylor0  I agree with David. The solution should be independent of acks enabled or not. In case of Sqs, the progress check is needed only in case of acks enabled because when acks are not enabled, the sqs message is deleted immediately. It looks like this solution is a hack.

Reply:
Ownership tracking, renewal, cancellation etc should be part of the source coordinator itself, and not in S3 code. Isn't this issue possible in other sources that use source coordinator?